ocaml/xenopsd/cli/xn.ml

```diff
@@ -1175,27 +1175,38 @@ let raw_console_proxy sockaddr =
       ) else if !final then
         finished := true
       else
-        let r, _, _ = Unix.select [Unix.stdin; fd] [] [] (-1.) in
-        if List.mem Unix.stdin r then (
-          let b =
-            Unix.read Unix.stdin buf_remote !buf_remote_end
-              (block - !buf_remote_end)
-          in
-          let i = ref !buf_remote_end in
-          while
-            !i < !buf_remote_end + b
-            && Char.code (Bytes.get buf_remote !i) <> 0x1d
-          do
-            incr i
-          done ;
-          if !i < !buf_remote_end + b then final := true ;
-          buf_remote_end := !i
-        ) ;
-        if List.mem fd r then
-          let b =
-            Unix.read fd buf_local !buf_local_end (block - !buf_local_end)
-          in
-          buf_local_end := !buf_local_end + b
+        let epoll = Polly.create () in
+        List.iter
+          (fun fd -> Polly.add epoll fd Polly.Events.inp)
+          [Unix.stdin; fd] ;
+        Fun.protect
+          ~finally:(fun () -> Polly.close epoll)
+          (fun () ->
+            ignore
+            @@ Polly.wait epoll 2 (-1) (fun _ file_desc _ ->
+                   if Unix.stdin = file_desc then (
+                     let b =
+                       Unix.read Unix.stdin buf_remote !buf_remote_end
+                         (block - !buf_remote_end)
+                     in
+                     let i = ref !buf_remote_end in
+                     while
+                       !i < !buf_remote_end + b
+                       && Char.code (Bytes.get buf_remote !i) <> 0x1d
+                     do
+                       incr i
+                     done ;
+                     if !i < !buf_remote_end + b then final := true ;
+                     buf_remote_end := !i
+                   ) ;
+                   if fd = file_desc then
```

Comment:
An alternative style could use `match`:

```
match file_desc with
| file_desc when file_desc = Unix.stdin -> ..
| file_desc when file_desc = fd -> ..
| _otherwise ->
```

Reply:
Do we want to handle one or all FDs that are ready? Up to `2` can be ready. This code handles two if they are ready but we have other places where we handle only one. It's not obvious why we have different behavior.